agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,609 @@
+	return CredentialSpecStatus(status).String()
+}
+
+// SetCreatedAt sets the timestamp for resource's creation time
+func (cs *CredentialSpecResource) SetCreatedAt(createdAt time.Time) {
+	if createdAt.IsZero() {
+		return
+	}
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	cs.createdAt = createdAt
+}
+
+// GetCreatedAt sets the timestamp for resource's creation time
+func (cs *CredentialSpecResource) GetCreatedAt() time.Time {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return cs.createdAt
+}
+
+// getRequiredCredentialSpecs returns the requiredCredentialSpecs field of credentialspec task resource
+func (cs *CredentialSpecResource) getRequiredCredentialSpecs() map[string][]*apicontainer.Container {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return cs.requiredCredentialSpecs
+}
+
+// getExecutionCredentialsID returns the execution role's credential ID
+func (cs *CredentialSpecResource) getExecutionCredentialsID() string {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return cs.executionCredentialsID
+}
+
+// GetName safely returns the name of the resource
+func (cs *CredentialSpecResource) GetName() string {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return ResourceName
+}
+
+// Create is used to create all the credentialspec resources for a given task
+func (cs *CredentialSpecResource) Create() error {
+	// To fail fast, check execution role first
+	executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
+	if !ok {
+		// No need to log here. managedTask.applyResourceState already does that
+		err := errors.New("credentialspec resource: unable to find execution role credentials")
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+	iamCredentials := executionCredentials.GetIAMRoleCredentials()
+
+	for credSpecStr, _ := range cs.requiredCredentialSpecs {
+		credSpecSplit := strings.SplitAfterN(credSpecStr, "credentialspec:", 2)
+		if len(credSpecSplit) != 2 {
+			seelog.Errorf("Invalid credentialspec: %v", credSpecStr)
+			continue
+		}
+		credSpecValue := credSpecSplit[1]
+
+		if strings.HasPrefix(credSpecValue, "file://") {
+			return cs.handleCredentialspecFile(credSpecStr)
+		}
+
+		parsedARN, err := arn.Parse(credSpecValue)
+		if err != nil {
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+
+		parsedARNService := parsedARN.Service
+		if parsedARNService == "s3" {
+			return cs.handleS3CredentialspecFile(credSpecStr, credSpecValue, iamCredentials)
+		} else if parsedARNService == "ssm" {
+			return cs.handleSSMCredentialspecFile(credSpecStr, credSpecValue, iamCredentials)
+		} else {
+			err := errors.New("unsupported credentialspec ARN, only s3/ssm ARNs are valid")
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) handleCredentialspecFile(credentialspec string) error {
+	credSpecSplit := strings.SplitAfterN(credentialspec, "credentialspec:", 2)
+	credSpecFile := credSpecSplit[1]
```

Comment:
same as previous comment, please check length before you perform array slicing. Can you please check for other occurrences

Reply:
done